naga/src/back/spv/writer.rs

```diff
@@ -167,7 +167,11 @@ impl Writer {
                 let selected = match self.capabilities_available {
                     None => first,
                     Some(ref available) => {
-                        match capabilities.iter().find(|cap| available.contains(cap)) {
+                        match capabilities
+                            .iter()
+                            // (need explicit type for hashbrown::HashSet::contains fn call to keep rustc happy)
+                            .find(|cap| available.contains::<spirv::Capability>(cap))
```

Comment:
OK I just added a "keep rustc happy" comment on this so we can move on. I have a feeling that this is related to how hashbrown uses an Equivalent trait, now wondering if it would be possible to simplify this on the hashbrown side. I am also wondering if this could be some kind of a compiler issue.

I tried annotating `|cap|`, VS Code suggests annotating with `&&spirv::Capability`, still need explicit type in the `hashbrown::HashSet::contains` call. I am not personally convinced to add this annotation as I think this should be clear enough from the explicit type in the contains call.

I think this part should be good to go, please let me know if you think otherwise about the changes here.